File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        result = [0, 0]

        for p in s:
            if p == '(':
                result[0] += 1
            elif p == ')' and result[0] > 0:
                result[0] -= 1
            else:
                result[1] += 1

        return sum(result)

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int[] result = new int[2];

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                result[0]++;
            } else if (result[0] > 0) {
                result[0]--;
            } else {
                result[1]++;
            }
        }

        return result[0] + result[1];
    }
}